packages/provider/src/providers/base-near.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Chain,
+  ChainId,
+  EthProviderRpcError,
+  NearProviderType,
+  NearProviderRpcError,
+  NearTransactionResponse,
+  NearTxRequestBody,
+  ProviderProxy,
+  RawProvider,
+  TransactionResponse,
+  TxRequestBody,
+} from '../types'
+import { HttpProvider, TransactionConfig } from 'web3-core'
+import Web3 from 'web3/types'
+import { NEAR_CHAINS } from '../enums'
+import {
+  getNearExplorerAddressUrl,
+  getNearExplorerTxUrl,
+  handleEthError,
+  handleNearError,
+} from '../helpers'
+
+import { ProviderEventBus } from './event-bus'
+
+export class BaseNear extends ProviderEventBus implements ProviderProxy {
+  readonly #provider: NearProviderType
+  readonly #web3: Web3
+
+  #isConnected = false
+  #chainId?: ChainId
+  #address?: string
+
+  constructor(provider: RawProvider) {
+    super()
+    this.#web3 = new window.Web3(provider as unknown as HttpProvider)
+    this.#provider = (<unknown>this.#web3?.currentProvider) as NearProviderType
+    this.#isConnected = false
```

Comment:
no need to make additional `#isConnected` field, in most cases `isConnected` is `Boolean(this.#chainId && this.#address)`